File: backend/Domain/Entities/Vehicle.cs
using SmartParking.Domain.Common;

namespace SmartParking.Domain.Entities;

public class Vehicle : BaseEntity
{
    public int VehicleID { get; set; }
    public string LicensePlate { get; set; } = string.Empty;

    public int UserID { get; set; }
    public User User { get; set; } = null!;

    public ICollection<Booking> Bookings { get; set; } = new List<Booking>();
    public ICollection<ParkingSession> Sessions { get; set; } = new List<ParkingSession>();
}

File: frontend/lib/models/vehicle.dart
class Vehicle {
  final int? vehicleID;
  final String licensePlate;
  final int? userID;

  const Vehicle({
    this.vehicleID,
    required this.licensePlate,
    this.userID,
  });

  factory Vehicle.fromJson(Map<String, dynamic> json) {
    return Vehicle(
      vehicleID: json['vehicleID'] as int?,
      licensePlate: json['licensePlate'] as String? ?? '',
      userID: json['userID'] as int?,
    );
  }
}